datajunction-server/datajunction_server/sql/parsing/ast.py

```diff
@@ -1043,7 +1052,7 @@ def add_ref_column(
                         column.add_type(col.type)
                         return True
 
-        for col in self.columns:
+        for col in self._columns:
```

Comment:
Why the switch to the obfuscated attribute? I think the property does some filtering in a case like:

```sql
SELECT column1, column2
FROM tableName(column1, column2)
```
where I think spark would only read `column1, column2` from the table and so no other columns would be legally referenced in the query despite maybe actually being in the table or node columns and seen during compiling the table with the node metadata.

Reply:
so if `tableName` has columns `column1, ...column3` then `.columns` will only show `column1, column2` but `._columns` would show them all.

Reply:
Good point - I'm not sure why I changed it, but it was likely just while I was testing something. I switched it back.